components/script/dom/document.rs

```diff
@@ -1654,16 +1654,21 @@ impl Document {
         );
         // TODO: Step 6, decrease the event loop's termination nesting level by 1.
         // Step 7
-        self.salvageable.set(!has_listeners);
+        if has_listeners {
+            self.salvageable.set(false);
+        }
         let mut can_unload = true;
         // TODO: Step 8 send a message to embedder to prompt user.
         // Step 9
         if !recursive_flag {
             for iframe in self.iter_iframes() {
                 // TODO: handle the case of cross origin iframes.
                 let document = document_from_node(&*iframe);
-                if !document.prompt_to_unload(true) {
-                    self.salvageable.set(document.salvageable());
+                can unload = document.prompt_to_unload(true);
+                if !document.salvageable() {
+                   self.salvageable.set(false);
+                }
+                if !can_unload {
```

Comment:
ended up changing this, because previously I would check `document.salvageable()` before calling `document.prompt_to_unload(true)`. Who thought checking booleans could be so hard...